Reaqtive/Core/Reaqtive.Linq/Reaqtive/Operators/ElementAt.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT License.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace Reaqtive.Operators
+{
+    internal sealed class ElementAt<TSource> : SubscribableBase<TSource>
+    {
+        private readonly ISubscribable<TSource> _source;
+        private readonly int _index;
+        private readonly bool _throwIfNotFound;
+
+        public ElementAt(ISubscribable<TSource> source, int index, bool throwIfNotFound)
+        {
+            Debug.Assert(source != null);
+            Debug.Assert(index >= 0);
+
+            _source = source;
+            _index = index;
+            _throwIfNotFound = throwIfNotFound;
+        }
+
+        protected override ISubscription SubscribeCore(IObserver<TSource> observer)
+        {
+            return new _(this, observer);
+        }
+
+        private sealed class _ : StatefulUnaryOperator<ElementAt<TSource>, TSource>, IObserver<TSource>
+        {
+            private int _currentIndex;
+
+            public _(ElementAt<TSource> parent, IObserver<TSource> observer)
+                : base(parent, observer)
+            {
+            }
+
+            public override string Name => "rc:ElementAt";
+
+            public override Version Version => Versioning.v1;
+
+            public void OnCompleted()
+            {
+                if (Params._throwIfNotFound)
+                {
+                    Output.OnError(new ArgumentOutOfRangeException("The element at the specified index was not found."));
+                }
+                else
+                {
+                    Output.OnNext(default);
+                    Output.OnCompleted();
+                }
+                Dispose();
+            }
+
+            public void OnError(Exception error)
+            {
+                Output.OnError(error);
+                Dispose();
+            }
+
+            public void OnNext(TSource value)
+            {
+                if (_currentIndex == Params._index)
+                {
+                    Output.OnNext(value);
+                    Output.OnCompleted();
+                    Dispose();
+                }
+                else
+                {
+                    checked
```

Comment:
The `checked` arithmetic is redundant in this case, because our `Params._index` is guaranteed to be in the range `[0, int.MaxValue)`, and `_currentIndex` starts off at `0`, so we'll bail out before we ever have a chance to overflow.

Note that classic Rx uses a countdown of an index value that starts off at the user-supplied index; when it hits `0`, the elements get emitted.

https://github.com/dotnet/reactive/blob/main/Rx.NET/Source/src/System.Reactive/Linq/Observable/ElementAt.cs#L40

which is very similar to the enumerable logic

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Linq/src/System/Linq/ElementAt.cs

I'm happy to count up here, and have the current index in the checkpoint state. In neither approach we do need `checked` arithmetic, so we can save a bit by removing that.

(FWIW, if there's any place where we need `checked`, we should always wrap it in a `try...catch...` in order to forward the `OverflowException` through `OnError`. See for example how `Select` deals with its `index++` logic.)